File: backend/app/services/water_service.py
from sqlalchemy.orm import Session
from app.models.water_log import WaterLog
from datetime import datetime, date

def create_water_log(user_id: int, amount_ml: float, db: Session) -> WaterLog:
    water_log = WaterLog(
        user_id=user_id,
        amount_ml=amount_ml
    )
    db.add(water_log)
    db.commit()
    db.refresh(water_log)
    return water_log

def get_water_logs_by_date(user_id: int, target_date: date, db: Session):
    start_of_day = datetime.combine(target_date, datetime.min.time())
    end_of_day = datetime.combine(target_date, datetime.max.time())
    
    logs = db.query(WaterLog).filter(
        WaterLog.user_id == user_id,
        WaterLog.logged_at >= start_of_day,
        WaterLog.logged_at <= end_of_day
    ).all()
    
    return logs

def get_total_water_by_date(user_id: int, target_date: date, db: Session) -> float:
    logs = get_water_logs_by_date(user_id, target_date, db)
    return sum(log.amount_ml for log in logs)

def delete_water_log(log_id: int, user_id: int, db: Session) -> bool:
    water_log = db.query(WaterLog).filter(
        WaterLog.id == log_id,
        WaterLog.user_id == user_id
    ).first()
    
    if not water_log:
        return False
    
    db.delete(water_log)
    db.commit()
    return True

def update_water_log(log_id: int, user_id: int, amount_ml: float, db: Session) -> WaterLog:
    water_log = db.query(WaterLog).filter(
        WaterLog.id == log_id,
        WaterLog.user_id == user_id
    ).first()
    
    if not water_log:
        return None
    
    water_log.amount_ml = amount_ml
    db.commit()
    db.refresh(water_log)
    return water_log

File: frontend/src/services/waterService.js
import api from './api';

const waterService = {
  // יצירת רישום מים חדש
  createWaterLog: async (amount_ml) => {
    const response = await api.post('/api/v1/water/', { amount_ml });
    return response.data;
  },

  // קבלת כל רישומי המים לתאריך מסוים
  getWaterLogs: async (date, clientId = null) => {
    let url = `/api/v1/water/logs?date=${date}`;
    if (clientId) url += `&client_id=${clientId}`;
    const response = await api.get(url);
    return response.data;
  },

  // קבלת סה"כ מים לתאריך מסוים
  getTotalWater: async (date, clientId = null) => {
    let url = `/api/v1/water/total?date=${date}`;
    if (clientId) url += `&client_id=${clientId}`;
    const response = await api.get(url);
    return response.data;
  },

  // מחיקת רישום מים
  deleteWaterLog: async (logId) => {
    const response = await api.delete(`/api/v1/water/${logId}`);
    return response.data;
  },

  // עדכון רישום מים
  updateWaterLog: async (logId, amount_ml) => {
    const response = await api.put(`/api/v1/water/${logId}`, { amount_ml });
    return response.data;
  },
};

export default waterService;